src/agent/debuglet.ts

```diff
@@ -106,6 +112,44 @@ const formatBreakpoints = function(
           .join('\n');
 };
 
+/**
+ * CachedPromise class is designed to support debug agent in Google Cloud
+ * Function (GCF). As GCF is a serverless environment and we wanted to make
+ * sure debug agent could always capture the snapshots, we designed this class
+ * so that GCF could wait until the listActiveBreakpoints event is completed
+ * before throttling CPU to 0. CachedPromise has two member functions.
+ * get() is to create or return an existing promise. resolve() will resolve
+ * the promise at appropriate code location (in our case is after
+ * listActiveBreakpoints is executed).
+ */
+export class CachedPromise {
+  private promise: Promise<void>|undefined;
+  private promiseResolve: (() => void)|null;
+  private promiseResolvedTimestamp = -Infinity;
+  private refresh: number;
```

Comment:
The name of this could be clearer... maybe `timeUntilStaleMS`?

Reply:
Done.